components/widgets/separator/separator.tsx

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {ReactNode} from 'react';
+import './separator.scss';
+
+type Props = {
+    className: string;
```

Comment:
Yep. You are correct. I'll refactor into two components.
Can you elaborte about: "By the way, I would expect to have a Separator class at the root node of the component to be able to reference the component from other parts of the css as a component."

Reply:
Sure, we are following a "policy" around CSS classes in the widgets library, each widget have a "root" class, and all the css related to that component is under that root class, the class name is the same as the component name, for example, something like:

for the js file
```js
MyNewWidget = (props) => (<div className='MyNewWidget'>...</div>);
```

For the scss file
```scss
.MyNewWidget {
  .classWhatever {
    ...
  }
  .otherClass {
    ...
  }
}
```

This way the css from the component doesn't affect in anyway nothing outside the component, if something outside the components wants to affect the component itself, still can, but we have to be careful there. If something outside the component wants to refer the components as a whole, it would use the component name as class name and thats it, for example:

```scss
.app-body {
    .center-channel {
        .MyNewWidget {
            ...
        }
    }
}

Reply:
Done